ui/app/ducks/gas/gas.duck.js

```diff
@@ -177,134 +176,132 @@ export function gasEstimatesLoadingFinished () {
 }
 
 export function fetchBasicGasEstimates () {
-  return (dispatch, getState) => {
-    const {
-      basicPriceEstimatesLastRetrieved,
-      basicPriceAndTimeEstimates,
-    } = getState().gas
+  return async (dispatch, getState) => {
+    const { basicPriceEstimatesLastRetrieved } = getState().gas
     const timeLastRetrieved = basicPriceEstimatesLastRetrieved || loadLocalStorageData('BASIC_PRICE_ESTIMATES_LAST_RETRIEVED') || 0
 
     dispatch(basicGasEstimatesLoadingStarted())
 
-    const promiseToFetch = Date.now() - timeLastRetrieved > 75000
-    ? fetch('https://dev.blockscale.net/api/gasexpress.json', {
-      'headers': {},
-      'referrer': 'https://dev.blockscale.net/api/',
-      'referrerPolicy': 'no-referrer-when-downgrade',
-      'body': null,
-      'method': 'GET',
-      'mode': 'cors'}
-    )
-      .then(r => r.json())
-      .then(({
-        safeLow,
-        standard: average,
-        fast,
-        fastest,
-        block_time: blockTime,
-        blockNum,
-      }) => {
-        const basicEstimates = {
-          safeLow,
-          average,
-          fast,
-          fastest,
-          blockTime,
-          blockNum,
-        }
-
-        const timeRetrieved = Date.now()
-        dispatch(setBasicPriceEstimatesLastRetrieved(timeRetrieved))
-        saveLocalStorageData(timeRetrieved, 'BASIC_PRICE_ESTIMATES_LAST_RETRIEVED')
-        saveLocalStorageData(basicEstimates, 'BASIC_PRICE_ESTIMATES')
-
-        return basicEstimates
-      })
-    : Promise.resolve(basicPriceAndTimeEstimates.length
-        ? basicPriceAndTimeEstimates
-        : loadLocalStorageData('BASIC_PRICE_ESTIMATES')
-      )
-
-    return promiseToFetch.then(basicEstimates => {
-      dispatch(setBasicGasEstimateData(basicEstimates))
-      dispatch(basicGasEstimatesLoadingFinished())
-      return basicEstimates
-    })
+    let basicEstimates
+    if (Date.now() - timeLastRetrieved > 75000) {
+      basicEstimates = await fetchExternalBasicGasEstimates(dispatch)
+    } else {
+      const cachedBasicEstimates = loadLocalStorageData('BASIC_PRICE_ESTIMATES')
+      basicEstimates = cachedBasicEstimates || await fetchExternalBasicGasEstimates(dispatch)
```

Comment:
I strongly considered falling back to `gas.basicEstimates` here instead (it certainly would have meant less refactoring), but I was concerned about the case where a call was made recently (and saved in `BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`), but `gas.basicEstimates` was still in it's initial state.

I could update this to check whether `gas.basicEstimates` was initialized or not, and use that instead of local storage or this additional network call.

It's not entirely clear to me why there was never a fallback to `gas.basicEstimates` here in the first place - maybe the intention was to _always_ update this value, for some reason?

Reply:
I it seems likely that the intention of the removed code
```
Promise.resolve(basicPriceAndTimeEstimates.length
        ? basicPriceAndTimeEstimates
        : loadLocalStorageData('BASIC_PRICE_ESTIMATES')
      )
```
was to fall back to `gas.basicEstimates` but it utilized `basicPriceAndTimeEstimates` by mistake.

I think that "I could update this to check whether gas.basicEstimates was initialized or not, and use that instead of local storage or this additional network call" is  good approach.

To be clear, the specifications we want to meet are:
- only make a request to the provided APIs at most once every 75 seconds
- never show data that is more than 75 seconds old
- if showing cached data, always show the most recently cached data

Reply:
> I it seems likely that the intention of the removed code ... was to fall back to `gas.basicEstimates`

Ahh, that makes sense. Thanks!

I'll go ahead with updating this PR to use `gas.basicEstimates` if it has been initialized then. That should be a strict improvement over what's here currently.

Reply:
Wait... nevermind. Perhaps not.

`gas.basicEstimates` is being used for two separate things right now: basic price estimates, and basic price *and time* estimates. They are stored separately in local storage, but they're stored on the same state variable. Fetching one overwrites whatever was there before.

The current behaviour of fetching from local storage at least ensures that, if the appropriate fetch function is called before use, the correct estimates will be set in `gas.basicEstimates`. If we fallback to `gas.estimates` we'll be preventing this overwrite behaviour, which means `gas.basicEstimates` would be "stuck" with either the basic price or the basic price and time estimates.

So I could either split them into two separate variables in state, or treat `BASIC_GAS_AND_TIME_API_ESTIMATES` as a strict superset of `BASIC_PRICE_ESTIMATES`. The latter option would be a little more complex in terms of the logic needed here, but it might be a simpler change to make overall because I wouldn't have to trace everywhere this state is used, and decide which copy should be used in each place. I'm also not _entirely_ sure that it's a strict superset. It looks like it could be, but the data is coming from separate APIs so I'm not inclined to assume.

Thoughts? I'm leaning toward the first option, though that can be in a separate PR.